pkg/utils/concurrency.go

```diff
@@ -0,0 +1,110 @@
+package utils

```

Comment:
similarly, probably all of this should go under its own `types` package instead of `utils`

Reply:
Slightly playing around with the name - this and ErrorOr will be moved to either `pkg/types` or `pkg/concurrency`, but together.

If you've got a strong opinion on which, I'm more than happy to go that way

Reply:
Since utils was absolutely wrong, and I needed to extract those functions out of the for loops I went for `pkg/concurency`

Leaving this comment open for now in case you'd prefer `pkg/types`